environmentManager/index.js

```diff
@@ -0,0 +1,73 @@
+const settings = require('../utils/parserConfigs')();
+const { validEnvironment } = require('../utils/parserConfigs/validators');
+const { getSplitFactory } = require('../sdk');
+
+class EnvironmentManager {
+  
+  constructor() {
+    
+    // In environments are the apiKey, factory an clientReadiness status related to an authToken
+    this._environments = {};
+    
+    // Ready promises for each client in environment manager
+    this._readyPromises = [];
+    
+    this._initializeEnvironments(JSON.parse(process.env['SPLIT_EVALUATOR_ENVIRONMENTS']));
```

Comment:
Is `process.env['SPLIT_EVALUATOR_ENVIRONMENTS']` always defined at this point? Otherwise, calling `JSON.parse(undefined)` will throw an error with a confusing message for the user.
Maybe you should handle it and throw a more descriptive error message.

Reply:
I added a `_getConfig` method that uses `nullOrEmpty` to validate that SPLIT_EVALUATOR_ENVIRONMENTS is always defined at that point